src/types.rs

```diff
@@ -474,6 +486,24 @@ impl Square {
         };
     }
 
+    /// Once you close a door, it CANNOT be re-opened via open_door()
```

Comment:
You could enfroce this further by making `open_door` check whether the wall is Impassable. You can never open an impassable wall.

Reply:
I added the comment since it already is impossible - because open_door will only open doors that are in one of DOOR_TYPES which doesn't include  Impassable. could add comment explaining that?